http-server-netty/src/main/java/io/micronaut/http/server/netty/jackson/JsonContentProcessor.java

```diff
@@ -78,91 +84,78 @@ public HttpContentProcessor resultType(Argument<?> type) {
-                if (out == null) {
-                    throw new IllegalStateException("Concurrent access not allowed");
-                }
-                failure = t;
-            }
-
-            @Override
-            protected void doOnComplete() {
-                if (out == null) {
-                    throw new IllegalStateException("Concurrent access not allowed");
-                }
-            }
-        });
-        this.jacksonProcessor.onSubscribe(new Subscription() {
-            @Override
-            public void request(long n) {
-            }
-
-            @Override
-            public void cancel() {
-                // happens on error, ignore
-            }
-        });
         return this;
     }
 
     @Override
     protected void onData(ByteBufHolder message, Collection<Object> out) throws Throwable {
-        if (jacksonProcessor == null) {
-            resultType(null);
-        }
-
-        this.out = out;
         ByteBuf content = message.content();
         try {
-            byte[] bytes = ByteBufUtil.getBytes(content);
-            jacksonProcessor.onNext(bytes);
+            countLoop(out, content);
+        } catch (Exception e) {
+            if (this.buffer != null) {
+                this.buffer.release();
+                this.buffer = null;
+            }
+            throw e;
         } finally {
-            ReferenceCountUtil.release(content);
-            this.out = null;
+            content.release();
         }
-        Throwable f = failure;
-        if (f != null) {
-            failure = null;
-            throw f;
+    }
+
+    private void countLoop(Collection<Object> out, ByteBuf content) throws IOException {
+        long initialPosition = counter.position();
+        long bias = initialPosition - content.readerIndex();
+        while (content.isReadable()) {
+            counter.feed(content);
+            JsonCounter.BufferRegion bufferRegion = counter.pollFlushedRegion();
+            if (bufferRegion != null) {
+                long start = Math.max(initialPosition, bufferRegion.start());
+                flush(out, content.retainedSlice(
+                    Math.toIntExact(start - bias),
+                    Math.toIntExact(bufferRegion.end() - start)
+                ));
+            }
+        }
+        if (counter.isBuffering()) {
+            int currentBufferStart = Math.toIntExact(Math.max(initialPosition, counter.bufferStart()) - bias);
+            bufferForNextRun(content.retainedSlice(currentBufferStart, content.writerIndex() - currentBufferStart));
         }
     }
 
-    @Override
-    public void complete(Collection<Object> out) throws Throwable {
-        if (jacksonProcessor == null) {
-            resultType(null);
+    private void bufferForNextRun(ByteBuf buffer) {
+        if (this.buffer == null) {
+            // number of components should not be too small to avoid unnecessary consolidation
+            this.buffer = buffer.alloc().compositeBuffer(4096);
```

Comment:
should this buffer size be configurable?

Reply:
can't hurt, though i didnt see a disadvantage to large values in testing. i will add a config